framework/doc/content/source/meshgenerators/MeshLineCutter.md

```diff
@@ -0,0 +1,75 @@
+
+The methods can be used to by this mesh generator to achieve the mesh cutting: the "Element Cutting" (`CUT_ELEM`) method and the "Node Moving" (`MOV_NODE`) method, which can be selected through [!param](/Mesh/MeshLineCutter/cutting_type). The `MOV_NODE` method is the same method used by `HexagonMeshTrimmer` and `CartesianMeshTrimmer` in the `Reactor` module. That is, the nodes near the cutting line are moved to the cutting line to ensure the smoothness of the cross-section. On the other hand, the `CUT_ELEM` method first converts all the QUAD4 elements crossed by the cutting line into pairs of TRI3 elements. Then all the TRI3 elements crossed by the cutting line are further split into two or three TRI elements with a division matching the cutting line.
+
+The two methods have their own Cons and Pros, respectively, and are complementary to each other. The `CUT_ELEM` method keeps all the subdomain interfaces and even preserved the volume of the subdomains. However, it may leads to some extraneously fine elements near the cutting line. On the other hand, the `MOV_NODE` method does not produce too fine elements but may deform the subdomain interfaces. Therefore, if only one subdomain is involved in the cutting, or all the cut subdomain interfaces are symmetric over the cutting line, the `MOv_NODE` method is preferred. On the other hand, of complex subdomain interfaces are involved in the cutting, the `CUT_ELEM` method is preferred.
+
+The id of the new external boundary produced by the trimming can be assigned using [!param](/Mesh/MeshLineCutter/new_boundary_id). As either method creates TRI3 elements if there are QUAD4 elements crossed by the cutting line, and as EXODUS II format does not allow TRI3 and QUAD4 elements sharing the same subdomain id and name. The newly formed TRI3 subdomains' ids and names can be customized using [!param](/Mesh/MeshLineCutter/tri_elem_subdomain_name_suffix) and [!param](/Mesh/MeshLineCutter/tri_elem_subdomain_shift).
+
+## Example Syntax
+
+The simple example of this mesh generator is to cut a square mesh that contains only one subdomain as shown in [simple_cut_original].
+
+!media mesh_modifiers/simple_cut_before.png
+      style=display: block;margin-left:auto;margin-right:auto;width:40%;
+      id=simple_cut_original
+      caption=The simple one-block mesh to be trimmed by this mesh generator.
+
+The simple mesh shown above can be cut by `MeshLineCutter` with `CUT_ELEM` method using the following input block:
+
+!listing test/tests/meshgenerators/mesh_line_cutter/simple_cut.i block=Mesh/mlc
+
+The mesh after the line cutting using `CUT_ELEM` method is shown in [simple_cut_elem]
+
```

Comment:
looking at those, it looks like in most cases it would better than both strategies to:

if one node is going to be cut out, just make a new node at the cut and use it for the QUAD that is getting cut (no moving a node from inside the domain)
if two nodes, same, just make two new nodes on the line
if three nodes, make two new nodes and degenerate from a QUAD to a TRI

Reply:
CUT_ELEM would be that strategy imo, what we have as CUT_ELEM is more SPLIT_THEN_CUT

Reply:
Cutting a QUAD4 element has many if/else's, that's why I use "split then cut" here.

It may be easier to recombine elements after cutting. Let me look into this to see what is the best approach.

For the node moving part, I should be able to make it better (but maybe not perfect).